Update libmesh

loganharbour opened the pull request:
Summary of changes:

 - Adds PETSc 3.19 compatability fixes
 - Makes linear partitioner idempotent for distributed meshes
 - Updates TIMPI to add sync command line switch
 - Allows edges with different orientations to match in ExodusII_IO
 - Removes overzealous assertion in copy_nodes_and_elements
 - Makes use of isspace more robust
 - Fixes NumericVector::compare
 - Fixes make check with MetaPhysical without MPI

loganharbour committed:
Update libmesh

Summary of changes:
- Adds PETSc 3.19 compatability fixes
- Makes linear partitioner idempotent for distributed meshes
- Updates TIMPI to add sync command line switch
- Allows edges with different orientations to match in ExodusII_IO
- Removes overzealous assertion in copy_nodes_and_elements
- Makes use of isspace more robust
- Fixes NumericVector::compare
- Fixes make check with MetaPhysical without MPI

refs #0

loganharbour committed:
Add CLI option to change TIMPI sync type

refs #24091

loganharbour committed:
Update versioner hashes due to libmesh update in #24092

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1466289/) on 93b505f wanted to post the following:

A change of the following file(s) triggered this check:

  conda/libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1466289/) on 93b505f wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

lindsayad reviewed (changes requested):
We are waiting on @roystgnr's constraint work aren't we?

roystgnr committed:
Update libMesh MORE!

Summary of changes:

- Update TIMPI; simplify sync CLI option, better enforce no empty sends
- Mesh equality tests, MeshTools::valid_is_prepared() test
- Close vector before adding with it (fixes Newmark solves with PETSc)
- Don't send empty elem request vectors
- Apply constraints via element nonlinear residuals
- Update TIMPI submodule (for more NBX error checking, code cleanup)
- Fix warnings with newer compilers

roystgnr committed:
Newsletter entries

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 80494219d2293e467392189814d30330c4f87a25
+Subproject commit 5e720a226ba3740dbe92c135936612461f489ada
```
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1470120/) on 69ab8fd wanted to post the following:

A change of the following file(s) triggered this check:

  conda/libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1470120/) on 69ab8fd wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1470120/) on 69ab8fd wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1470119/) on 69ab8fd wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24092/site)

This comment will be updated on new commits.

lindsayad reviewed (approved):
so happy
